test/org/sagebionetworks/bridge/models/schedules/ABTestScheduleStrategyTest.java

```diff
@@ -26,4 +71,86 @@ public void testScheduleCollector() {
         assertEquals("Schedule 3", schedules.get(2).getLabel());
         assertTrue(schedules instanceof ImmutableList);
     }
+    
+    @Test
+    public void canRountripABTestingPlan() throws Exception {
+        DynamoSchedulePlan plan = createABSchedulePlan();
+        String output = new BridgeObjectMapper().writeValueAsString(plan);
+
+        JsonNode node = mapper.readTree(output);
+        DynamoSchedulePlan newPlan = DynamoSchedulePlan.fromJson(node);
+
+        assertEquals("Plan with AB testing strategy was serialized/deserialized", plan, newPlan);
+
+        ABTestScheduleStrategy strategy = (ABTestScheduleStrategy) plan.getStrategy();
+        ABTestScheduleStrategy newStrategy = (ABTestScheduleStrategy) newPlan.getStrategy();
+        assertEquals("Deserialized AB testing strategy is complete", strategy.getScheduleGroups().get(0).getSchedule(),
+                        newStrategy.getScheduleGroups().get(0).getSchedule());
+    }
+
+    @Test
+    public void verifyABTestingStrategyWorks() {
+        DynamoSchedulePlan plan = createABSchedulePlan();
+
+        List<Schedule> schedules = Lists.newArrayList();
+        for (User user : users) {
+            ScheduleContext context = new ScheduleContext.Builder().withUser(user).build();
+            Schedule schedule = plan.getStrategy().getScheduleForUser(plan, context);
+            schedules.add(schedule);
+        }
+
+        // We want 4 in A, 4 in B and 2 in C
+        // and they should not be in order...
+        Map<String, Integer> countsByLabel = Maps.newHashMap();
+        for (Schedule schedule : schedules) {
+            Integer count = countsByLabel.get(schedule.getLabel());
+            if (count == null) {
+                countsByLabel.put(schedule.getLabel(), 1);
+            } else {
+                countsByLabel.put(schedule.getLabel(), ++count);
+            }
+        }
```

Comment:
Minor: A Guava Multiset will make this much simpler https://code.google.com/p/guava-libraries/wiki/NewCollectionTypesExplained#Multiset